src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectory.Stream.Tests.cs

```diff
@@ -264,5 +264,170 @@ public void UnseekableStreams_RoundTrip(TestTarFormat testFormat)
+            ExtractToDirectory_ExactRootDirMatch_RegularFile_And_Directory_Throws_Internal(format, entryType, fileName, inverted: true);
+        }
+
+        [Fact]
+        public void ExtractToDirectory_ExactRootDirMatch_Directory_Relative_Throws()
+        {
+            string entryFolderName = "folder";
+            string destinationFolderName = "folderSibling";
+
+            using TempDirectory root = new TempDirectory();
+
+            string entryFolderPath = Path.Join(root.Path, entryFolderName);
+            string destinationFolderPath = Path.Join(root.Path, destinationFolderName);
+
+            Directory.CreateDirectory(entryFolderPath);
+            Directory.CreateDirectory(destinationFolderPath);
+
+            // Relative segments should not change the final destination folder
+            string dirPath1 = Path.Join(entryFolderPath, "..", "folder");
+            string dirPath2 = Path.Join(entryFolderPath, "..", "folder" + Path.DirectorySeparatorChar);
+
+            ExtractRootDirMatch_Verify_Throws(TarEntryFormat.Ustar, TarEntryType.Directory, destinationFolderPath, dirPath1, linkTargetPath: null);
+            ExtractRootDirMatch_Verify_Throws(TarEntryFormat.Ustar, TarEntryType.Directory, destinationFolderPath, dirPath2, linkTargetPath: null);
+        }
+
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.SupportsHardLinkCreation))]
+        [InlineData(TarEntryFormat.V7)]
+        [InlineData(TarEntryFormat.Ustar)]
+        [InlineData(TarEntryFormat.Pax)]
+        [InlineData(TarEntryFormat.Gnu)]
+        public void ExtractToDirectory_ExactRootDirMatch_HardLinks_Throws(TarEntryFormat format)
+        {
+            ExtractToDirectory_ExactRootDirMatch_Links_Throws(format, TarEntryType.HardLink, inverted: false);
+            ExtractToDirectory_ExactRootDirMatch_Links_Throws(format, TarEntryType.HardLink, inverted: true);
+        }
+
+        [ConditionalTheory(typeof(MountHelper), nameof(MountHelper.CanCreateSymbolicLinks))]
+        [InlineData(TarEntryFormat.V7)]
+        [InlineData(TarEntryFormat.Ustar)]
+        [InlineData(TarEntryFormat.Pax)]
+        [InlineData(TarEntryFormat.Gnu)]
+        public void ExtractToDirectory_ExactRootDirMatch_SymLinks_Throws(TarEntryFormat format)
+        {
+            ExtractToDirectory_ExactRootDirMatch_Links_Throws(format, TarEntryType.SymbolicLink, inverted: false);
+            ExtractToDirectory_ExactRootDirMatch_Links_Throws(format, TarEntryType.SymbolicLink, inverted: true);
+        }
+
+        [ConditionalFact(typeof(MountHelper), nameof(MountHelper.CanCreateSymbolicLinks))]
+        public void ExtractToDirectory_ExactRootDirMatch_SymLinks_TargetOutside_Throws()
+        {
+            string entryFolderName = "folder";
+            string destinationFolderName = "folderSibling";
+
+            using TempDirectory root = new TempDirectory();
```

Comment:
Possibly creating temp directory is cleaner in a single GlobalSetup?

Just looking for ways to make these tests shorter and cleaner by removing repeated stuff